src/main/java/com/smilestore/service/UserService.java

```diff
@@ -0,0 +1,86 @@
+package com.smilestore.service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.smilestore.controller.ProductController;
+import com.smilestore.exception.UserException;
+import com.smilestore.model.Role;
+import com.smilestore.model.User;
+import com.smilestore.repository.UserRepository;
+
+@Service
+public class UserService implements UserDetailsService {
+	
+	private static final Logger LOGGER = Logger.getLogger(ProductController.class);
```

Comment:
Why is ProductController Logger used?
And please use in such cases `Logger.getLogger( ClassName.class.getName() );`